### PR TITLE
feat: rely on revm implementation of EIP-2537

### LINKED ISSUE #74: Rely on revm implementation of EIP-2537
Now that we have an implementation of EIP-2537 in revm, we should import that code now instead. There were also many bug fixes for EIP-2537 included in revm which should apply here as well.

### PR DESCRIPTION
**Pull request by fgimenez**
Closes #74 

Bumps reth and revm dependencies, uses BLS implementation from revm and removes its local benches, test vectors and gh bench workflow.

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
nice!

File: Cargo.toml
Comment (Rjected) [inline Cargo.toml:69]: We probably want to either patch or point to a rev, since the BLS fixes have not been released yet:

https://github.com/bluealloy/revm/compare/v36...main

**Review by Rjected — APPROVED** [review]
LGTM, just one point about BLS fixes in revm

File: Cargo.toml
Comment (fgimenez) [reply Cargo.toml:69]: @Rjected great point! the problem with both the patch and pointing to a rev is that we have revm that comes from reth and revm imported with the `blst` feature, if the revs are not the same  `reth::revm::precompile::PrecompileWithAddress` and `revm::precompile::PrecompileWithAddress` no longer match. wdyt, should we wait for a new revm release?

Comment (fgimenez) [reply Cargo.toml:69]: we can also merge as is and bump revm when there's a new release, this would unblock the 3074 refactor

Comment (fgimenez) [reply Cargo.toml:69]: to be more precise, 3074 refactor is not blocked by these changes, but some of the changes here would be beneficial for it, we can move them to the 3074 refactor work and wait for revm here.

Comment (Rjected) [reply Cargo.toml:69]: hmm, does it still not work since the reth bump here? we don't need to wait, and can just update revm later if so
https://github.com/paradigmxyz/reth/pull/8447/files

Comment (fgimenez) [reply Cargo.toml]: yep, it works with the latest bump, thx!

**Pull request merged**
